RealmSwift/Map.swift

```diff
@@ -0,0 +1,600 @@
+     */
+    public func max() -> Value? {
+        return _rlmCollection.max(ofProperty: "self").map(dynamicBridgeCast)
+    }
+}
+
+extension Map where Value: OptionalProtocol, Value.Wrapped: MinMaxType {
+    /**
+     Returns the minimum (lowest) value of the map, or `nil` if the map is empty.
+     */
+    public func min() -> Value.Wrapped? {
+        return _rlmCollection.min(ofProperty: "self").map(dynamicBridgeCast)
+    }
+    /**
+     Returns the maximum (highest) value of the map, or `nil` if the map is empty.
+     */
+    public func max() -> Value.Wrapped? {
+        return _rlmCollection.max(ofProperty: "self").map(dynamicBridgeCast)
+    }
+}
+
+extension Map where Value: AddableType {
+    /**
+     Returns the sum of the values in the map.
+     */
+    public func sum() -> Value {
+        return sum(ofProperty: "self")
+    }
+
+    /**
+     Returns the average of the values in the map, or `nil` if the map is empty.
+     */
+    public func average<T: AddableType>() -> T? {
+        return average(ofProperty: "self")
+    }
+}
+
+public extension Map where Value: OptionalProtocol, Value.Wrapped: AddableType {
+    /**
+     Returns the sum of the values in the map, or `nil` if the map is empty.
+     */
+    func sum() -> Value.Wrapped {
+        return sum(ofProperty: "self")
+    }
+    /**
+     Returns the average of all of the values in the collection.
+     */
+    func average<T: AddableType>() -> T? {
+        return average(ofProperty: "self")
+    }
+}
+
+// MARK: - AssistedObjectiveCBridgeable
+
+extension Map: AssistedObjectiveCBridgeable {
+    internal static func bridging(from objectiveCValue: Any, with metadata: Any?) -> Map {
+        guard let objectiveCValue = objectiveCValue as? RLMDictionary<AnyObject, AnyObject> else { preconditionFailure() }
+        return Map(objc: objectiveCValue)
+    }
+
+    internal var bridged: (objectiveCValue: Any, metadata: Any?) {
+        return (objectiveCValue: _rlmCollection, metadata: nil)
+    }
+}
+
+// MARK: Sequence Support
+
+extension Map: Sequence {
+    /// Returns a `RLMMapIterator` that yields successive elements in the `Map`.
+    public func makeIterator() -> RLMMapIterator<SingleMapEntry<Key, Value>> {
+        return RLMMapIterator(collection: rlmDictionary)
+    }
+}
+
+// MARK: - Notifications
+
+/**
+ A `RealmMapChange` value encapsulates information about changes to dictionaries
+ that are reported by Realm notifications.
+ */
+@frozen public enum RealmMapChange<Collection: RealmKeyedCollection> {
+
+    /**
+     `.initial` indicates that the initial run of the query has completed (if
+     applicable), and the collection can now be used without performing any
+     blocking work.
+     */
+    case initial(Collection)
+
+    /**
+     `.update` indicates that a write transaction has been committed which
+     either changed which objects are in the collection, and/or modified one
+     or more of the objects in the collection.
```

Comment:
"which keys are in the collection, or the values of the objects for those keys" or something like that.